### PR TITLE
feat: String Export Settings API implementation 

### PR DESCRIPTION
**Pull request by OmAximani0**
- refs: #99

### CODE REVIEW

**Comment by andrii-bodnar** [comment]
@OmAximani0 thank you, please include in this PR the rest API methods described in the issue

**Comment by OmAximani0** [comment]
Ok @andrii-bodnar , I'll do that

File: crowdin_api/api_resources/projects/resource.py [hunk 1]
```diff
@@ -372,3 +375,114 @@ def edit_project_file(
...
+    def get_strings_exporter_path(
+        self,
+        projectId: int,
+        systemStringExporterSettingId: Optional[int] = None
+    ):
+        if systemStringExporterSettingId is None:
+            return f"projects/{projectId}/strings-exporter-settings"
+        return f"projects/{projectId}/strings-exporter-settings/{systemStringExporterSettingId}"
+
+    def list_project_strings_exporter(
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:388]:
```suggestion
    def list_project_strings_exporter_settings(
```

File: crowdin_api/api_resources/projects/resource.py [hunk 2]
```diff
@@ -372,3 +375,114 @@ def edit_project_file(
...
+        self,
+        projectId: int,
+        page: Optional[int] = None,
+        offset: Optional[int] = None,
+        limit: Optional[int] = None,
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:393]: this method doesn't accept such parameters

File: crowdin_api/api_resources/projects/resource.py [hunk 3]
```diff
@@ -372,3 +375,114 @@ def edit_project_file(
...
+        https://developer.crowdin.com/api/v2/#operation/api.projects.strings-exporter-settings.getMany
+        """
+        params = self.get_page_params(page=page, offset=offset, limit=limit)
+        return self._get_entire_data(
+            method="get",
+            path=self.get_strings_exporter_path(projectId=projectId),
+            params=params,
+        )
+
+    def add_project_strings_exporter(
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:408]:
```suggestion
    def add_project_strings_exporter_settings(
```

File: crowdin_api/api_resources/projects/resource.py [hunk 4]
```diff
@@ -372,3 +375,114 @@ def edit_project_file(
...
+        Link to documetation:
+        https://developer.crowdin.com/api/v2/#operation/api.projects.strings-exporter-settings.post
+        """
+        return self.requester.request(
+            method="post",
+            path=self.get_strings_exporter_path(projectId=projectId),
+            request_data={"format": format, "settings": settings},
+        )
+
+    def get_project_strings_exporter(
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:430]:
```suggestion
    def get_project_strings_exporter_settings(
```

File: crowdin_api/api_resources/projects/resource.py [hunk 5]
```diff
@@ -372,3 +375,114 @@ def edit_project_file(
...
+        """
+        return self.requester.request(
+            method="get",
+            path=self.get_strings_exporter_path(
+                projectId=projectId,
+                systemStringExporterSettingId=systemStringsExporterSettingId,
+            ),
+        )
+
+    def delete_project_strings_exporter(
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:447]:
```suggestion
    def delete_project_strings_exporter_settings(
```

File: crowdin_api/api_resources/projects/resource.py [hunk 6]
```diff
@@ -372,3 +375,114 @@ def edit_project_file(
             ),
             request_data=data,
         )
+
+    def get_strings_exporter_path(
+        self,
+        projectId: int,
+        systemStringExporterSettingId: Optional[int] = None
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:382]:
```suggestion
        systemStringExporterSettingsId: Optional[int] = None
```

File: crowdin_api/api_resources/projects/resource.py [hunk 7]
```diff
@@ -372,3 +375,114 @@ def edit_project_file(
...
+        https://developer.crowdin.com/api/v2/#operation/api.projects.strings-exporter-settings.post
+        """
+        return self.requester.request(
+            method="post",
+            path=self.get_strings_exporter_path(projectId=projectId),
+            request_data={"format": format, "settings": settings},
+        )
+
+    def get_project_strings_exporter(
+        self, projectId: int, systemStringsExporterSettingId: int
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:431]:
```suggestion
        self, projectId: int, systemStringsExporterSettingsId: int
```

File: crowdin_api/api_resources/projects/resource.py [hunk 8]
```diff
@@ -372,3 +375,114 @@ def edit_project_file(
...
+        return self.requester.request(
+            method="get",
+            path=self.get_strings_exporter_path(
+                projectId=projectId,
+                systemStringExporterSettingId=systemStringsExporterSettingId,
+            ),
+        )
+
+    def delete_project_strings_exporter(
+        self, projectId: int, systemStringsExporterSettingId: int
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:448]:
```suggestion
        self, projectId: int, systemStringsExporterSettingsId: int
```

File: crowdin_api/api_resources/projects/resource.py [hunk 9]
```diff
@@ -372,3 +375,114 @@ def edit_project_file(
...
+        Delete Project Strings Exporter Settings.
+
+        Link to documetation:
+        https://developer.crowdin.com/api/v2/#operation/api.projects.strings-exporter-settings.delete
+        """
+        return self.requester.request(
+            method="delete",
+            path=self.get_strings_exporter_path(
+                projectId=projectId,
+                systemStringExporterSettingId=systemStringsExporterSettingId,
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:460]: `systemStringExporterSettingId` -> `systemStringExporterSettingsId`

please fix this everywhere in the code

**Review by andrii-bodnar — COMMENTED** [review]
Please also add Unit tests for the new methods

https://github.com/crowdin/crowdin-api-client-python/blob/main/crowdin_api/api_resources/projects/tests/test_projects_resources.py

File: crowdin_api/api_resources/projects/resource.py (hunk 2 above)
Comment (OmAximani0) [reply crowdin_api/api_resources/projects/resource.py:393]: For this I looked at `list_project_files` and  there are params like this and response for both are same
https://github.com/crowdin/crowdin-api-client-python/blob/3b598b9a6795b36de3bfea43d40fdee77d266938/crowdin_api/api_resources/projects/resource.py#L278-L284

Comment (andrii-bodnar) [reply crowdin_api/api_resources/projects/resource.py:393]: @OmAximani0 please remove

File: crowdin_api/api_resources/projects/resource.py [hunk 10]
```diff
@@ -372,3 +375,109 @@ def edit_project_file(
...
+        Get Project Strings Exporter Settings
+
+        Link to documetation:
+        https://developer.crowdin.com/api/v2/#operation/api.projects.strings-exporter-settings.get
+        """
+        return self.requester.request(
+            method="get",
+            path=self.get_strings_exporter_path(
+                projectId=projectId,
+                systemStringExporterSettingId=systemStringExporterSettingsId,
```
Comment (andrii-bodnar) [inline crowdin_api/api_resources/projects/resource.py:438]:
```suggestion
                systemStringExporterSettingsId=systemStringExporterSettingsId,
```

**Review by andrii-bodnar — APPROVED** [review]
@OmAximani0 thank you!

**Comment by andrii-bodnar** [comment]
@OmAximani0 I merged this PR without noticing that the Unit tests were not added (https://github.com/crowdin/crowdin-api-client-python/pull/105#pullrequestreview-1628425334)

Could you please do it via a separate PR?

**Comment by OmAximani0** [comment]
> @OmAximani0 I merged this PR without noticing that the Unit tests were not added ([#105 (review)](https://github.com/crowdin/crowdin-api-client-python/pull/105#pullrequestreview-1628425334))
> 
> Could you please do it via a separate PR?

Yes, I will open a new PR for that. If ok you can create a new issue where I can reference it

**Comment by andrii-bodnar** [comment]
@OmAximani0 please reference just the current PR

**Pull request merged**